### ISSUE TITLE
Refactor CMake usage to best practices

### ISSUE THREAD

**Issue by ronaldtse** [rnpgp/rnp#1542]
The setup instructions of `googletest` suggests usage of the CMake `FetchContent` commands, but we're not doing that right now:
https://google.github.io/googletest/quickstart-cmake.html

We would want to take this chance to refactor the CMake configuration and build flow using current best practices.

**Comment by ronaldtse** [comment]
Ping @thierryseegers

**Comment by thierryseegers** [comment]
@ronaldtse Am I right in thinking that you'd ultimately like to get rid of as much workarounds/complexity as possible? I'm referring to these (there may be more):
- `GTEST_SOURCES` and whatever CMake code related to it.
- `DOWNLOAD_GTEST` and whatever CMake code related to it.
- The workaround for MSYS I see in `./github/workflows/windows.yml`.

**Comment by ronaldtse** [comment]
@thierryseegers indeed, the less complexity we have, the easier to maintain. Thanks!

**Comment by ni4** [comment]
JFYI: GTEST_SOURCES/DOWNLOAD_GTEST were added in support of offline builds, please see https://github.com/rnpgp/rnp/issues/1410 for discussion.

**Comment by ronaldtse** [comment]
Thanks @ni4 for the reminder... let's see if we can simplify given this constraint of #1410.

**Comment by thierryseegers** [comment]
@ronaldtse After a simple conversion of `ExternalProject` to `FetchContent`, a chain of annoyances ensued: 
1. When building the target of `FetchContent`, the current compilation options and flags will apply, in the case of an MSVC build, it picks up `/std:c++latest` from the root `CMakeLists.txt`. This fails compilation of `@c43f710` specified in the original `gtest-CMakeLists.txt.in`. It's a single compilation error in a header.
2. Curiously, on other platforms, `gmock` gets built when it appears it wasn't with the old way of doing things. I'm not positive I understand why. That also fails compilation with a single compilation error.

Issue no 2 is circumvented by forcing `googletest`'s `BUILD_GMOCK` option to `OFF`.

Issue no 1 is proving more complicated. Upgrading `googletest` to `release-1.11.0` (the latest) fixes the compilation problem for MSVC but brings in a new one on CentOS7. It appears that `std::is_trivially_copy_constructible` is not part of the libstdc++ found on CentOS7. From a cursory investigation, it looks like `/std:c++latest` may be needed only for one feature: designated initializers of structs (some examples found in `ecdh.cpp,50:57`).

Thus the crossroads it appears I'm finding myself at is:
- Revert to googletest version `@c43f710`, remove `/std:c++latest` and fix whatever compilation errors occur.
- I'm not sure what else because I don't have an idea how to fix the compilation error observed on CentOS7.

For info, [here](https://github.com/thierryseegers/rnp/pull/1) are my CI results. There's a failure observed for MSVC/x64 but it's a timeout for the `cli_tests-Compression` test which I'm assuming is spurious.

**Comment by ronaldtse** [comment]
@thierryseegers thanks for #1547. Could you help with the rest of CMake usage and then CPack? Thanks.
